src/PrestaShopBundle/Command/SecurityAnnotationLinterCommand.php

```diff
@@ -27,35 +27,155 @@
 namespace PrestaShopBundle\Command;
 
 use PrestaShopBundle\Routing\Linter\Exception\LinterException;
+use PrestaShopBundle\Routing\Linter\SecurityAnnotationLinter;
 use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Input\InputArgument;
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Output\OutputInterface;
 use Symfony\Component\Console\Style\SymfonyStyle;
 use Symfony\Component\Routing\Route;
 
 /**
  * Checks if all admin routes have @AdminSecurity configured
+ *
+ * @see \PrestaShopBundle\Security\Annotation\AdminSecurity
  */
 final class SecurityAnnotationLinterCommand extends ContainerAwareCommand
 {
+    const ACTION_LIST_ALL = 'list';
+    const ACTION_FIND_MISSING = 'find-missing';
+
+    /**
+     * @param string $expression
+     *
+     * @return string
+     */
+    public static function parseExpression($expression)
+    {
+        $pattern1 = '#\[(.*)\]#';
+        $pattern2 = '#is_granted\((.*),#';
+        $matches1 = [];
+        $matches2 = [];
```

Comment:
```suggestion
```

Reply:
You don't need to declare these variables :thinking:

Reply:
Not declaring variables 🤔? I use them after to be given back the matches by `preg_match` . Even if it's possible, I think tools like phpstan will tell us it's not good

Reply:
Phpstan will not trigger any errors because variables passed by reference are automatically created by PHP.